src/context/ChannelContext.tsx

```diff
@@ -0,0 +1,249 @@
+    colons: string;
+    id: string;
+    name: string;
+    sheet_x: number;
+    sheet_y: number;
+  };
+
+export type EmojiConfig = {
+  commonEmoji: CommonEmoji;
+  defaultMinimalEmojis: MinimalEmoji[];
+  Emoji: React.ComponentType<NimbleEmojiProps>;
+  emojiData: EmojiMartData;
+  EmojiIndex: NimbleEmojiIndex;
+  EmojiPicker: React.ComponentType<NimblePickerProps>;
+  emojiSetDef: {
+    sheetColumns: number;
+    sheetRows: number;
+    sheetSize: number;
+    size: number;
+    spriteUrl: string;
+  };
+};
+
+export type MessageToSend<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = {
+  attachments?: Array<
+    | Attachment<At>
+    | {
+        fallback: string;
+        image_url: string | undefined;
+        type: string;
+      }
+    | {
+        asset_url: string;
+        file_size: number;
+        mime_type: string;
+        title: string;
+        type: string;
+      }
+  >;
+  id?: string;
+  mentioned_users?: string[];
+  parent?: MessageResponse<At, Ch, Co, Me, Re, Us>;
+  text?: string;
+};
+
+export type ChannelState<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = {
+  error?: Error | null;
+  hasMore?: boolean;
+  loading?: boolean;
+  loadingMore?: boolean;
+  members?: StreamChannelState<At, Ch, Co, Ev, Me, Re, Us>['members'];
+  messages?: StreamChannelState<At, Ch, Co, Ev, Me, Re, Us>['messages'];
+  pinnedMessages?: StreamChannelState<
+    At,
+    Ch,
+    Co,
+    Ev,
+    Me,
+    Re,
+    Us
+  >['pinnedMessages'];
+  read?: StreamChannelState<At, Ch, Co, Ev, Me, Re, Us>['read'];
+  thread?: ReturnType<
+    StreamChannelState<At, Ch, Co, Ev, Me, Re, Us>['formatMessage']
+  > | null;
+  threadHasMore?: boolean;
+  threadLoadingMore?: boolean;
+  threadMessages?: Array<
+    ReturnType<StreamChannelState<At, Ch, Co, Ev, Me, Re, Us>['formatMessage']>
+  >;
+  typing?: StreamChannelState<At, Ch, Co, Ev, Me, Re, Us>['typing'];
+  watcherCount?: number;
+  watchers?: StreamChannelState<At, Ch, Co, Ev, Me, Re, Us>['watchers'];
+};
+
+export type ChannelContextValue<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = ChannelState<At, Ch, Co, Ev, Me, Re, Us> & {
+  client: StreamChat<At, Ch, Co, Ev, Me, Re, Us>;
```

Comment:
why is client on ChannelContext? we should keep it only in ChatContext. it'll get confusing if we're listening to events on 2 potentially different clients

Reply:
don't know why it's passed in there, but didn't want to pull out and potentially cause a break. figured we'd just rip out when we reorg context after TS